grammar/FuncTestCaseParser.g4

```diff
@@ -0,0 +1,227 @@
+parser grammar FuncTestCaseParser;
+
+options {
+    caseInsensitive = true;
+    tokenVocab=SubstraitLexer;
+    tokenVocab=FuncTestCaseLexer;
+}
+
+doc
+    : header testGroup+ EOF
+    ;
+
+header
+    : version include
+    ;
+
+version
+    : SubstraitScalarTest FormatVersion
+    ;
+
+include
+    : SubstraitInclude StringLiteral (Comma StringLiteral)*
+    ;
+
+testGroupDescription
+    : DescriptionLine
+    ;
+
+testCase
+    : functionName=Identifier OParen arguments CParen ( OBracket func_options CBracket )? Eq result
+    ;
+
+testGroup
+    : testGroupDescription (testCase)+
+    ;
+
+arguments
+    : argument (Comma argument)*
+    ;
+
+result
+    : argument
+    | substraitError
+    ;
+
+argument
+    : nullArg
+    | i8Arg | i16Arg | i32Arg | i64Arg
+    | fp32Arg | fp64Arg
+    | booleanArg
+    | stringArg
+    | decimalArg
+    | dateArg
+    | timeArg
+    | timestampArg
+    | timestampTzArg
+    | intervalYearArg
+    | intervalDayArg
+    ;
+
+numericLiteral
+    : DecimalLiteral | IntegerLiteral | floatLiteral
+    ;
+
+floatLiteral
+    : FloatLiteral | NaN
+    ;
+
+nullArg: NullLiteral DoubleColon datatype;
+
+i8Arg: IntegerLiteral DoubleColon I8;
```

Comment:
Is there any value to having i8Arg separate from any other integer -- no matter what happens you will need to check that the value fits the chosen size.

For that matter, the same applies to other numbers.  Another benefit is that instead of someone being told that they need to put an i8/i16/i132/i64/fp32/fp64 when they have a syntax error is that it would tell them they need a number.

Reply:
It may help to add tests for syntax errors so that behavior can be seen.

Reply:
Added tests. Combined few rules.